test/Microsoft.AspNetCore.Mvc.FunctionalTests/HtmlGenerationTest.cs

```diff
@@ -554,6 +556,45 @@ public class HtmlGenerationTest :
             Assert.Contains("Hello, Private World!", response);
         }
 
+        [Fact]
+        public async Task PartialTagHelper_AllowsPassingModelValue()
+        {
+            // Arrange
+            var url = "/HtmlGeneration_Home/StatusMessage";
+
+            // Act
+            var document = await Client.GetHtmlDocumentAsync(url);
+
+            // Assert
+            var banner = QuerySelector(document, ".banner");
+            Assert.Equal("Some status message", banner.TextContent);
+        }
+
+        [Fact]
+        public async Task PartialTagHelper_AllowsPassingNullModelValue()
+        {
+            // Arrange
+            var url = "/HtmlGeneration_Home/NullStatusMessage";
+
+            // Act
+            var document = await Client.GetHtmlDocumentAsync(url);
+
+            // Assert
+            var banner = QuerySelector(document, ".banner");
+            Assert.Empty(banner.TextContent);
+        }
+
+        private static IElement QuerySelector(IHtmlDocument document, string selector)
+        {
+            var element = document.QuerySelector(selector);
+            if (element == null)
+            {
+                throw new ArgumentException($"Document does not contain element that matches the selector {selector}: " + Environment.NewLine + document.DocumentElement.OuterHtml);
```

Comment:
Why not `Assert.NotNull(element)`?

Reply:
It doesn't do a very good job of indicating what went wrong when the expectation isn't met. I'd like to write [Razor-style asserts](https://github.com/aspnet/Razor/blob/dev/test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/Assert.cs) for this, but that can wait until mq.